### PR TITLE
Converts Actor base class to a mixin

### PR DESCRIPTION
**Pull request by paullewis**
The changes are in Actor.ts and Actor_test.ts. Everything else is a build artifact.

### CODE REVIEW

File: src/actor/Actor.ts [hunk 1]
```diff
@@ -42,6 +42,122 @@ declare global {
...
+ * application, you can use `actorName`. This field is accessible only after
+ * the {@link hookup} has been called.
+ *
+ * Users of this mixin generally should not use {@link Actor#initPromise}. This
+ * is an internal implementation detail for {@link hookup}.
+ */
+export function actorMixin<T, S extends Constructable = Constructable<Object>>(
+  superClass: S
+) {
+  class actor extends superClass {
```
Comment (TimvdLippe) [inline src/actor/Actor.ts]: Nit: can be `return class extends superClass`

File: lib/actor/Actor.d.ts [hunk 2]
```diff
@@ -73,64 +81,65 @@ export declare type ValidActorMessageName = keyof ActorMessageType;
...
+export declare function actorMixin<T, S extends Constructable = Constructable<Object>>(superClass: S): {
+    new (...args: any[]): {
+        /**
+         * Do not use, it is an internal implementation detail used in {@link hookup}.
+         */
+        readonly initPromise: Promise<void>;
+        /**
+         * The name given to this actor by calling {@link hookup}.
+         */
+        actorName?: "ignoring" | "ignoring1" | "late" | undefined;
```
Comment (TimvdLippe) [inline lib/actor/Actor.d.ts]: This seems wrong, but I fear it is a TS issue rather than an "us" issue.

File: src/actor/Actor.ts (hunk 1 above)
Comment (paullewis) [reply src/actor/Actor.ts]: Done.

File: lib/actor/Actor.d.ts (hunk 2 above)
Comment (paullewis) [reply lib/actor/Actor.d.ts]: Yeah, didn't write this myself. Assuming `tsc` is the culprit.

Comment (TimvdLippe) [reply lib/actor/Actor.d.ts]: Linked to upstream issue https://github.com/Microsoft/TypeScript/issues/27171

**Pull request merged**